lib/data/gateways/sembast_database.dart

```diff
@@ -118,9 +137,9 @@ class SembastDatabaseImpl implements SembastDatabase {
 
   /// Transforms a single `sembast` snapshot record into an object
   final snapshotTransformer =
-      StreamTransformer<RecordSnapshot<String, Map<String, Object?>>, Map<String, Object?>>.fromHandlers(
+      StreamTransformer<RecordSnapshot<String, Map<String, Object?>>?, Map<String, Object?>?>.fromHandlers(
     handleData: (snapshot, sink) {
-      sink.add(snapshot.value);
+      sink.add(snapshot?.value);
```

Comment:
That's because it retrieves an object from the database, and it may be simply `null`. The listeners are the ones that have to deal with it being null or not, and not the stream transformer, because there might be some scenarios where an object simply is deleted, so an stream that did return a non-null object may start returning a null object, and vice-versa